core/src/main/java/org/apache/accumulo/core/fate/zookeeper/MetaFateStore.java

```diff
@@ -551,30 +552,48 @@ public Stream<FateKey> list(FateKey.FateKeyType type) {
         .filter(fateKey -> fateKey.getType() == type);
   }
 
-  protected static class NodeValue {
+  private Deque<Repo<T>> createEmptyRepoDeque() {
+    return new ArrayDeque<>();
+  }
+
+  private Map<TxInfo,Serializable> createEmptyTxInfo() {
+    return new EnumMap<>(TxInfo.class);
+  }
+
+  protected static class FateData<T> {
     final TStatus status;
     final Optional<FateKey> fateKey;
     final Optional<FateReservation> reservation;
-
-    private NodeValue(byte[] serializedData) {
+    final Deque<Repo<T>> repoDeque;
```

Comment:
Thoughts about using Deque/ArrayDeque for the repo stack?

Reply:
I like it.